### PR TITLE
Some update to the docs

### PR DESCRIPTION
**Pull request by francoishernandez**
This PR will track updates made to the docs.

### CODE REVIEW

**Comment by francoishernandez** [comment]
Note: the quickstart build_vocab command considers `-src_vocab` is not required, we'll need to make sure it's implemented.

**Comment by francoishernandez** [comment]
Note: the pretrained embeddings FAQ entry is based on #19

**Comment by francoishernandez** [comment]
Note: we'll need to add a link to the Transforms doc section in the tokenization FAQ question when it'll be done.

**Comment by francoishernandez** [comment]
Probably as a side effect of importing all transforms opt in both `build_vocab` and `train`, all options for all transforms are shown in both sections of the doc:
Screenshot from 2020-09-21 17-39-45
Screenshot from 2020-09-21 17-39-36

Not very bad, it's just unnecessary duplication. Not sure about the best way to fix this though.

**Comment by francoishernandez** [comment]
Note: we can remove src_lang/tgt_lang from example configs, since they are not required after #18 changes.

**Comment by francoishernandez** [comment]
# TODO
- [x] adapt library example following changes in #21 
- [x] there may be some changes following #22 ~~and #20~~ as well
- [x] some FAQ entry about how to add a custom transform
- [x] some FAQ entry about the transforms already available

**Comment by francoishernandez** [comment]
It seems the common tokenize options (Transform/Subword/Common) from the TokenizeTransform class are not properly grabbed when building the doc. Not sure where it comes from. Any idea @Zenglinxiao ?
Screenshot from 2020-09-25 09-55-25

**Comment by Zenglinxiao** [comment]
> It seems the common tokenize options (Transform/Subword/Common) from the TokenizeTransform class are not properly grabbed when building the doc. Not sure where it comes from. Any idea @Zenglinxiao ?

Maybe due to the `_COMMON_OPT_ADDED` flag used in the `TokenizeTransform` I guess?
But not sure how to easily eliminate this issue.

**Comment by francoishernandez** [comment]
> Maybe due to the _COMMON_OPT_ADDED flag used in the TokenizeTransform I guess?
But not sure how to easily eliminate this issue.

That's annoying but it's not that bad, I made an entry in the FAQ detailing the available transforms anyways.

File: README.md [hunk 1]
```diff
@@ -3,30 +3,70 @@
 [![Build Status](https://travis-ci.org/OpenNMT/OpenNMT-py.svg?branch=master)](https://travis-ci.org/OpenNMT/OpenNMT-py)
 [![Run on FH](https://img.shields.io/badge/Run%20on-FloydHub-blue.svg)](https://floydhub.com/run?template=https://github.com/OpenNMT/OpenNMT-py)
```
Comment (Zenglinxiao) [inline README.md]: Maybe add a badge link directly to the documentation site?
Looks like many other repo do so, ref. fairseq, huggingface, etc.
Also wondering if we can set up a coverage test CI to encourage more unitests making sure the change won't break some module.

File: README.md [hunk 2]
```diff
@@ -3,30 +3,70 @@
...
-an open-source (MIT) neural machine translation system. It is designed to be research friendly to try out new ideas in translation, summary, image-to-text, morphology, and many other domains. Some companies have proven the code to be production ready.
+This is the [PyTorch](https://github.com/pytorch/pytorch) version of the [OpenNMT](https://opennmt.net) project, an open-source (MIT) neural machine translation framework. It is designed to be research friendly to try out new ideas in translation, summary, morphology, and many other domains. Some companies have proven the code to be production ready.
 
 We love contributions. Please consult the Issues page for any [Contributions Welcome](https://github.com/OpenNMT/OpenNMT-py/issues?q=is%3Aissue+is%3Aopen+label%3A%22contributions+welcome%22) tagged post. 
 
 <center style="padding: 40px"><img width="70%" src="http://opennmt.github.io/simple-attn.png" /></center>
-
 Before raising an issue, make sure you read the requirements and the documentation examples.
 
 Unless there is a bug, please use the [Forum](http://forum.opennmt.net) or [Gitter](https://gitter.im/OpenNMT/OpenNMT-py) to ask questions.
```
Comment (Zenglinxiao) [inline README.md:16]: This can also be added to the badge section just as OpenNMT-tf.

File: README.md [hunk 3]
```diff
@@ -3,30 +3,70 @@
...
+Feel free to check it out and let us know what you think of the new paradigm!
+
+----
+
 
 Table of Contents
 =================
   * [Full Documentation](http://opennmt.net/OpenNMT-py/)
-  * [Requirements](#requirements)
+  * [Usage](#usage)
```
Comment (Zenglinxiao) [inline README.md:60]: Since all the text in this section is actually installing required packages, I prefer requirements in this case though.

File: README.md [hunk 4]
```diff
@@ -42,18 +82,16 @@ python setup.py install
...
 ```
 
-Note:
-
-- some features require Python 3.5 and after (eg: Distributed multigpu, entmax)
-- we currently only support PyTorch 1.4
-
 ## Features
 
+- **/!\ New in OpenNMT-py 2.0**: [On the fly data processing]([here](https://opennmt.net/OpenNMT-py/FAQ.html#what-are-the-readily-available-on-the-fly-data-transforms).)
```
Comment (Zenglinxiao) [inline README.md]: /!\ rendered weird, maybe use an emoji as replacement to make it stand out?

Comment (Zenglinxiao) [reply README.md]: Following lines may need to be changed:

- [Conv2Conv convolution model]
- SRU "RNNs faster than CNN" paper

File: README.md [hunk 5]
```diff
@@ -72,58 +110,117 @@ Note:
...
+# Prevent overwriting existing files in the folder
+overwrite: False
+
+# Corpus opts:
+data:
+    corpus_1:
+        path_src: toy-ende/src-train.txt
+        path_tgt: toy-ende/tgt-train.txt
+        transforms: []
+        weight: 1
```
Comment (Zenglinxiao) [inline README.md]: We can ignore these two if we don't set.

File: README.md [hunk 6]
```diff
@@ -156,6 +253,7 @@ Major contributors are:
 [Guillaume Klein](https://github.com/guillaumekln) (Systran)
 [Paul Tardy](https://github.com/pltrdy) (Ubiqus / Lium)
 [François Hernandez](https://github.com/francoishernandez) (Ubiqus)
+[Linxiao Zeng](https://github.com/Ubiqus/Zenglinxiao) (Ubiqus)
```
Comment (Zenglinxiao) [inline README.md]: Wrong url.

**Comment by francoishernandez** [comment]
Final TODOs:
- [x] update src/tgt tokenization options in FAQ
- [x] check yaml parameter configs
- [x] finish translation example, add data_prepare script somewhere
- [x] add badge to the docs at top of readme
- [x] add badge to gitter / forum at top of readme

File: docs/source/extended.md [hunk 7]
```diff
@@ -1,10 +1,10 @@
 
 # Translation
 
-This example is for training for the WMT'16 Multimodal Translation task (http://www.statmt.org/wmt16/multimodal-task.html). It will use on the fly tokenization with [pyonmttok](https://github.com/OpenNMT/Tokenizer) and [sacrebleu](https://github.com/mjpost/sacrebleu) for evaluation.
+This example is for training for the WMT'14 English to German news translation task (https://www.statmt.org/wmt14/translation-task.html). It will use on the fly tokenization with [sentencepiece](https://github.com/google/sentencepiece) and [sacrebleu](https://github.com/mjpost/sacrebleu) for evaluation.
```
Comment (Zenglinxiao) [inline docs/source/extended.md]: Can be better?
WMT'14 English to German news translation task (https://www.statmt.org/wmt14/translation-task.html) -> [WMT'14 English to German news translation task](https://www.statmt.org/wmt14/translation-task.html)

**Review by Zenglinxiao — COMMENTED** [review]
reviewed README, FAQ, extended.

File: README.md (hunk 1 above)
Comment (francoishernandez) [reply README.md]: Why not for the badge, I'll add it.
Coverage is a good idea. It may require a bit more time though. We'll add it later.

File: README.md (hunk 2 above)
Comment (francoishernandez) [reply README.md:16]: Will do.

File: docs/source/CONTRIBUTING.md [hunk 8]
```diff
@@ -5,7 +5,7 @@ OpenNMT-py is a community developed project and we love developer contributions.
 ## Guidelines
 Before sending a PR, please do this checklist first:
 
-- Please run `tools/pull_request_chk.sh` and fix any errors. When adding new functionality, also add tests to this script. Included checks:
+- Please run `onmt/tests/pull_request_chk.sh` and fix any errors. When adding new functionality, also add tests to this script. Included checks:
```
Comment (Zenglinxiao) [inline docs/source/CONTRIBUTING.md:8]: Some things that we can do later though, I find the `pull_request_chk.sh` run the same cmd in `.travis.yml`, just wonder if we can call the script directly from `.travis.yml` rather than copy-paste.

File: README.md (hunk 3 above)
Comment (francoishernandez) [reply README.md:60]: I'll go for "Setup" I think, because it's not only requirements, as we clone and install opennmt.

File: README.md (hunk 4 above)
Comment (francoishernandez) [reply README.md]: I don't think these are broken by the new codepaths, are they?

File: README.md (hunk 6 above)
Comment (francoishernandez) [reply README.md]: Copy paste mishap, my bad.

File: docs/source/extended.md (hunk 7 above)
Comment (francoishernandez) [reply docs/source/extended.md]: Indeed, forgot the brackets.

File: README.md (hunk 4 above)
Comment (Zenglinxiao) [reply README.md]: I mean we may add link to it if we use [] or term "paper".

File: docs/source/Summarization.md [hunk 9]
```diff
@@ -25,49 +25,53 @@ An example article-title pair from Gigaword should look like this:
...
+# common vocabulary for source and target
+share_vocab: True
+
+# Corpus opts:
+data:
+    cnndm:
+        path_src: cnndm/train.txt.src
+        path_tgt: cnndm/train.txt.tgt.tagged
+        transforms: []
+        weight: 1
```
Comment (Zenglinxiao) [inline docs/source/Summarization.md]: can be removed

File: docs/source/CONTRIBUTING.md (hunk 8 above)
Comment (francoishernandez) [reply docs/source/CONTRIBUTING.md:8]: Yes, we can probably do this at the same time we'll look at the coverage CI stuff.

File: README.md (hunk 4 above)
Comment (francoishernandez) [reply README.md]: Oh, right. Will do.

File: docs/source/quickstart.md [hunk 10]
```diff
@@ -18,37 +26,97 @@ The data consists of parallel source (`src`) and target (`tgt`) data containing
...
+# Prevent overwriting existing files in the folder
+overwrite: False
+
+# Corpus opts:
+data:
+    corpus_1:
+        path_src: toy-ende/src-train.txt
+        path_tgt: toy-ende/tgt-train.txt
+        transforms: []
+        weight: 1
```
Comment (Zenglinxiao) [inline docs/source/quickstart.md]: same

File: docs/source/quickstart.md [hunk 11]
```diff
@@ -18,37 +26,97 @@ The data consists of parallel source (`src`) and target (`tgt`) data containing
...
-The main train command is quite simple. Minimally it takes a data file
-and a save file.  This will run the default model, which consists of a
-2-layer LSTM with 500 hidden units on both the encoder/decoder.
-If you want to train on GPU, you need to set, as an example:
-CUDA_VISIBLE_DEVICES=1,3
-`-world_size 2 -gpu_ranks 0 1` to use (say) GPU 1 and 3 on this node only.
-To know more about distributed training on single or multi nodes, read the FAQ section.
+This configuration will run the default model, which consists of a 2-layer LSTM with 500 hidden units on both the encoder and decoder. It will run on a single GPU (`world_size 1` & `gpu_ranks [0]`).
+
+Before the training process actually starts, the `*.vocab.pt` together with `*.transforms.pt` will be dumpped to `-save_data` with configurations specified in `-config` yaml file. We'll also generate transformed samples to simplify any potentially required visual inspection. The number of sample lines to dump per corpus is set with the `-n_sample` flag.
```
Comment (Zenglinxiao) [inline docs/source/quickstart.md]: Not always dumpped anymore.

File: docs/source/refs.bib [hunk 12]
```diff
@@ -481,3 +481,21 @@ @inproceedings{DeeperTransformer
     pages = "1810--1822",
     abstract = "Transformer is the state-of-the-art model in recent machine translation evaluations. Two strands of research are promising to improve models of this kind: the first uses wide networks (a.k.a. Transformer-Big) and has been the de facto standard for development of the Transformer system, and the other uses deeper language representation but faces the difficulty arising from learning deep networks. Here, we continue the line of research on the latter. We claim that a truly deep Transformer model can surpass the Transformer-Big counterpart by 1) proper use of layer normalization and 2) a novel way of passing the combination of previous layers to the next. On WMT{'}16 English-German and NIST OpenMT{'}12 Chinese-English tasks, our deep system (30/25-layer encoder) outperforms the shallow Transformer-Big/Base baseline (6-layer encoder) by 0.4-2.4 BLEU points. As another bonus, the deep model is 1.6X smaller in size and 3X faster in training than Transformer-Big.",
 }
+
```
Comment (Zenglinxiao) [inline docs/source/refs.bib:484]: Relate to reference, as we delete the image, audio, video, should we delete also the references relate to them?

File: onmt/dynamic/opts.py [hunk 13]
```diff
@@ -7,23 +7,25 @@
 def _dynamic_corpus_opts(parser, build_vocab_only=False):
     """Options related to training corpus, type: a list of dictionary."""
     group = parser.add_argument_group('Data')
```
Comment (Zenglinxiao) [inline onmt/dynamic/opts.py:9]: I wrongly rechange the " to ' when fix merge confilct. Maybe fix that back?
Also is there any reason that we prefer " to ' ?

File: onmt/dynamic/opts.py [hunk 14]
```diff
@@ -7,23 +7,25 @@
...
 
-    group.add('-save_data', '--save_data', required=build_vocab_only,
-              help='Output directory for data saving path.')
+    group.add("-save_data", "--save_data", required=build_vocab_only,
+              help="Output base path for objects that will "
+                   "be saved (vocab, transforms, embeddings, ...).")
+    group.add("-overwrite", "--overwrite", action="store_true",
+              help="Overwrite existing objects if any.")
     group.add(
         '-n_sample', '--n_sample',
```
Comment (Zenglinxiao) [inline onmt/dynamic/opts.py:30]: same for '.

File: onmt/dynamic/opts.py [hunk 15]
```diff
@@ -47,43 +49,45 @@ def _dynamic_fields_opts(parser, build_vocab_only=False):
...
+        group.add("--src_seq_length_trunc", "-src_seq_length_trunc",
                   type=int, default=None,
                   help="Truncate source sequence length.")
-        group.add('--tgt_seq_length_trunc', '-tgt_seq_length_trunc',
+        group.add("--tgt_seq_length_trunc", "-tgt_seq_length_trunc",
                   type=int, default=None,
                   help="Truncate target sequence length.")
 
-        group = parser.add_argument_group('embeddings')
+        group = parser.add_argument_group('Embeddings')
```
Comment (Zenglinxiao) [inline onmt/dynamic/opts.py:90]: same

**Review by Zenglinxiao — COMMENTED** [review]
Brefly viewed all changes.

File: docs/source/refs.bib (hunk 12 above)
Comment (francoishernandez) [reply docs/source/refs.bib:484]: Still present in legacy. Does not hurt to keep them.

File: onmt/dynamic/opts.py (hunk 13 above)
Comment (francoishernandez) [reply onmt/dynamic/opts.py:9]: Nope, just wanted to make things homogeneous at some point, but it's quite useless.

File: docs/source/quickstart.md (hunk 11 above)
Comment (francoishernandez) [reply docs/source/quickstart.md]: Will update.

File: docs/source/Summarization.md (hunk 9 above)
Comment (francoishernandez) [reply docs/source/Summarization.md]: Will do.

**Comment by francoishernandez** [comment]
I think we can merge this one. Finish what needs to in #10, and we'll do the last changes to the docs at the end.
